src/components/HoldPages/HoldRequestErrorBanner.tsx

```diff
@@ -0,0 +1,177 @@
+import { useContext } from "react"
+import {
+  Text,
+  Box,
+  Banner,
+  Button,
+  List,
+} from "@nypl/design-system-react-components"
+
+import type {
+  HoldErrorStatus,
+  PatronEligibilityStatus,
+} from "../../types/holdPageTypes"
+import { FeedbackContext } from "../../context/FeedbackContext"
+import type { ItemMetadata } from "../../types/itemTypes"
+import type Item from "../../models/Item"
+import RCLink from "../Links/RCLink/RCLink"
+import ExternalLink from "../Links/ExternalLink/ExternalLink"
+import {
+  PATHS,
+  HOLD_PAGE_ERROR_HEADINGS,
+  HOLD_PAGE_CONTACT_PREFIXES,
+} from "../../config/constants"
+import { appConfig } from "../../config/config"
+
+interface HoldRequestErrorBannerProps {
+  item: Item
+  errorStatus?: HoldErrorStatus
+  patronEligibilityStatus?: PatronEligibilityStatus
+}
+
+/**
+ * The HoldRequestErrorBanner renders an error notification on the hold page that includes a button to
+ * open the feedback form, pre-populated with item metadata.
+ */
+const HoldRequestErrorBanner = ({
+  item,
+  errorStatus = "failed",
+  patronEligibilityStatus,
+}: HoldRequestErrorBannerProps) => {
+  const { onOpen, setItemMetadata } = useContext(FeedbackContext)
+
+  const onContact = (metadata: ItemMetadata) => {
+    setItemMetadata(metadata)
+    onOpen()
+  }
+
+  return (
+    <Banner
+      type="negative"
+      heading={HOLD_PAGE_ERROR_HEADINGS?.[errorStatus] || null}
```

Comment:
Does this work if it's both dot notation and bracket?

Reply:
Yeah it allows you to have a null check on an attribute with quotes (helpful if there's a digit or something) but it's not necessary here.

Reply:
Actually, it is necessary because it's a dynamic key.